### PR TITLE
[Docs] Configuring http proxy settings

### PR DESCRIPTION
**Pull request by bashofmann**
#### Proposed Changes ####

Add documentations on configuring http proxy settings.

#### Types of Changes ####

Docs

#### Linked Issues ####

Addresses https://github.com/rancher/rke2/issues/2091

### CODE REVIEW

**Review by Oats87 — APPROVED** [review]

**Review by brandond — APPROVED** [review]
Hmm. I thought that we added the cluster CIDRs and dns suffix to the NO_PROXY variable, but it looks like for some reason we only currently do that on server nodes, not agents. Until we fix that, this docs update should cover it.

File: docs/advanced.md [hunk 1]
```diff
@@ -22,6 +22,23 @@ For advanced customization of this file you can create another file called `conf
 
 The `config.toml.tmpl` will be treated as a Go template file, and the `config.Node` structure is being passed to the template. See [this template](https://github.com/k3s-io/k3s/blob/master/pkg/agent/templates/templates.go#L16-L32) for an example of how to use the structure to customize the configuration file.
 
+## Configuring an HTTP proxy for containerd
```
Comment (brandond) [inline docs/advanced.md:25]:
```suggestion
## Configuring an HTTP proxy
```

File: docs/advanced.md [hunk 2]
```diff
@@ -22,6 +22,23 @@ For advanced customization of this file you can create another file called `conf
...
+- `/etc/default/rke2-agent`
+
+The `NO_PROXY` variable must include your internal networks, as well as the cluster pod and service IP ranges.
+
+```
+HTTP_PROXY=http://your-proxy.example.com:8888
+HTTPS_PROXY=http://your-proxy.example.com:8888
+NO_PROXY=127.0.0.0/8,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.svc,.cluster.local
+```
+
```
Comment (brandond) [inline docs/advanced.md:41]: You can also prefix environment variables with CONTAINERD_ and they will be passed through to containerd only, with the prefix removed. If your intent is to set the proxy variables just for containerd, that's probably a better way to do it. Would you mind adding this to the documentation as well? The currently recommended variables will set it for both RKE2, the Kubelet, and containerd.

**Review by brandond — CHANGES_REQUESTED** [review]

File: docs/advanced.md (hunk 2 above)
Comment (bashofmann) [reply docs/advanced.md:41]: I added a section for this.

File: docs/advanced.md [hunk 3]
```diff
@@ -22,6 +22,31 @@ For advanced customization of this file you can create another file called `conf
...
+If you just want to configure the proxy settings for containerd to pull container images through the proxy, you can alsoe prefix these variables with `CONTAINERD_`:
```
Comment (brandond) [inline docs/advanced.md:42]:
```suggestion
If you want to configure the proxy settings for containerd without affecting RKE2 and the Kubelet, you can prefix the variables with `CONTAINERD_`:
```

**Review by brandond — CHANGES_REQUESTED** [review]
nit

**Review by brandond — APPROVED** [review]

**Pull request merged**